app/assets/javascripts/components/overview/milestones.jsx

```diff
@@ -28,19 +31,26 @@ const Milestones = createReactClass({
     const weekNumberOffset = CourseDateUtils.weeksBeforeTimeline(this.props.course);
     const blocks = [];
 
-    this.props.allWeeks.map((week) => {
+    
+    this.props.allWeeks.map((week,navIndex) => {
       if (week.empty) return null;
-
       const milestoneBlocks = filter(week.blocks, block => block.kind === this.milestoneBlockType);
       return milestoneBlocks.map((block) => {
+        if (
+          !this.props.weekDates[navIndex]?.start || 
+          !this.props.weekDates[navIndex]?.end || 
+          this.props.weekDates[navIndex]?.start > this.props.weekDates[navIndex]?.end
```

Comment:
This is quite verbose, but it's not obvious what this if statement is for. What situation(s) for a block are being checked for here?

Reply:
This check was added to prevent any date from showing after the timeline's end date in the milestone.

Reply:
Hmm... I don't think that' necessary. The Timeline can handle dates that are after the end date (in which case, they get highlighted in red). I think in the Milestones component, it would be better to show dates that stretch beyond the end date, when that happens.

Reply:
Okay, I will update the part where the deadline extends. Should I display the extended deadline within the milestone like in the below image, or should I show it after the timeline's end date?
![image](https://github.com/user-attachments/assets/985a3a33-cfbd-4018-bb52-26ab20b806fc)

Reply:
I don't think any special formatting is needed in the Milestones component.

Reply:
Yes, I have updated it. Could you please check it?